execution_engine_testing/test_support/src/wasm_test_builder.rs

```diff
@@ -941,6 +941,24 @@ where
         ret
     }
 
+    /// Gets all `[Key::Balance]`s in global state.
+    pub fn get_balances(&mut self) -> Vec<Key> {
+        let correlation_id = CorrelationId::new();
+        let state_root_hash = self.get_post_state_hash();
+
+        let tracking_copy = self
+            .engine_state
+            .tracking_copy(state_root_hash)
+            .unwrap()
+            .unwrap();
+
+        let reader = tracking_copy.reader();
+
+        reader
+            .keys_with_prefix(correlation_id, &[KeyTag::Withdraw as u8])
```

Comment:
The method says it will return balances' keys but here you're looking for a `Withdraw` key tag. There exists a `Key::Balance` tag so why not use it? I'm suspecting it's because you're actually not looking for balances but unbonding purses but if that's the case we should change the method name and its comment.

Reply:
No, that was a foolish mistake on my part, it is meant to be `Balance` cause that way I can track in my test if any new purses are being created. I re-ran the test with this correction and it seems to be working as intended.